src/routing.js

```diff
@@ -44,21 +44,32 @@ export const queryStringMiddleware = (history, {reduxPathname, routes}, config =
         const result = next(action);
         const state = store.getState();
         const currentPath = reduxPathname(state);
-        const routeConfig = routes[currentPath];
+        const routesConfig = !_.isArray(routes[currentPath]) ? [routes[currentPath]] : routes[currentPath];
 
-        if (routeConfig) {
-            const {listen, select, serialize} = routeConfig;
-            const qsState = select(state);
+        if (!_.isEmpty(routesConfig)) {
+            let queryString = [];
+            routesConfig.forEach((routeConfig) => {
```

Comment:
I believe you can use `.map()` instead of `.forEach()` here.
This is how I'd do it, actually:

```js
const queryString = routesConfig.filter(({select}) => { /* check if route matches */ })
                                .map(({serialize}) => { /* serialize data */ });
```

Reply:
Ah, I forgot you'd have to flatten the resulting structure:
```js
Object.assign(...data);
```